pennylane/templates/subroutines/qsvt.py

```diff
@@ -0,0 +1,233 @@
+# Copyright 2018-2023 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Contains the QSVT template and qsvt wrapper function.
+"""
+# pylint: disable=too-many-arguments
+import pennylane as qml
+from pennylane.queuing import QueuingManager
+from pennylane.ops import BlockEncode, PCPhase
+from pennylane.ops.op_math import adjoint
+from pennylane.operation import AnyWires, Operation
+from pennylane import numpy as np
+import copy
+
+
+def qsvt(A, angles, wires, convention=None):
+    r"""Performs the
+    `quantum singular value transformation <https://arxiv.org/abs/1806.01838>`__ using
+    :class:`~.BlockEncode` and :class:`~.PCPhase`.
+
+    Args:
+        A (tensor_like): the general :math:`(n \times m)` matrix to be encoded.
+        angles (tensor_like): a list of angles by which to shift to obtain the desired polynomial.
+        wires (Iterable): the wires the template acts on.
+        convenction (string): can be set to "Wx" to convert quantum signal processing angles to QSVT
+        angles
+
+    .. details::
+        :title: Usage Details
+
+        .. code-block:: python
+
+            dev = qml.device("default.qubit", wires=2)
+            A = [[0.1, 0.2], [0.3, 0.4]]
+            angles = [0.1, 0.2, 0.3]
+
+            @qml.qnode(dev)
+            def example_circuit(A):
+                qml.qsvt(A, angles, wires=[0, 1])
+                return qml.expval(qml.PauliZ(wires=0))
+
+            print(qml.draw(example_circuit)(A))
+
+            0: ─╭QSVT─┤  <Z>
+            1: ─╰QSVT─┤
+
+            print(qml.draw(example_circuit, expansion_strategy="device")(A))
+
+            0: ─╭BlockEncode(M0)─╭∏_ϕ(0.10)─╭BlockEncode(M0)†─╭∏_ϕ(0.20)─╭BlockEncode(M0)─╭∏_ϕ(0.30)─┤  <Z>
+            1: ─╰BlockEncode(M0)─╰∏_ϕ(0.10)─╰BlockEncode(M0)†─╰∏_ϕ(0.20)─╰BlockEncode(M0)─╰∏_ϕ(0.30)─┤
+    """
+    if qml.math.shape(A) == () or qml.math.shape(A) == (1,):
+        A = qml.math.reshape(A, [1, 1])
+
+    c, r = qml.math.shape(A)
+
+    UA = BlockEncode(A, wires=wires, do_queue=False)
+    projectors = []
+
+    if convention == "Wx":
+        angles = _qsp_to_qsvt(angles)
+        global_phase = (len(angles) - 1) % 4
+
+        if global_phase == 1:
```

Comment:
```suggestion
        if global_phase:
            qml.exp(qml.Identity(wires=wires), 0.5j * np.pi *  (4 - global_phase))
```
This should suffice instead of multiple if conditions.